lib/openfoodfacts.dart

```diff
@@ -436,14 +435,82 @@ class OpenFoodAPIClient {
 
   /// Uses the auth.pl API to see if login was successful
   /// Returns a bool if the login data of the provided user is correct
-  static Future<bool> login(User user) async {
+  static Future<bool> login(User user,
+      {QueryType queryType = QueryType.PROD}) async {
     var loginUri = Uri(
       scheme: URI_SCHEME,
-      host: URI_PROD_HOST,
+      host: queryType == QueryType.PROD ? URI_PROD_HOST : URI_TEST_HOST,
       path: '/cgi/auth.pl',
     );
     Response response =
         await HttpHelper().doPostRequest(loginUri, user.toData(), user);
     return response.statusCode == 200;
   }
+
+  /// Creates a new user
+  /// Returns [Status.status] 201 = complete; 400 = wrong inputs + [Status.error]; 500 = server error;
+  ///
+  /// When creating a [producer account](https://world.pro.openfoodfacts.org/) use [requested_org] to name the Producer or brand
+  static Future<Status> register({
+    required User user,
+    required String name,
+    required String email,
+    String? requested_org,
+    bool newsletter = true,
+    QueryType queryType = QueryType.PROD,
+  }) async {
+    var registerUri = Uri(
+      scheme: URI_SCHEME,
+      host: queryType == QueryType.PROD ? URI_PROD_HOST : URI_TEST_HOST,
+      path: '/cgi/user.pl',
+    );
+
+    Map<String, String> data = <String, String>{
+      'name': name,
+      'email': email,
+      'userid': user.userId,
+      'password': user.password,
+      'confirm_password': user.password,
+      if (requested_org != null) 'pro': 'on',
+      'pro_checkbox': '1',
+      'requested_org': requested_org ?? ' ',
+      if (newsletter) 'newsletter': 'on',
+      'action': 'process',
+      'type': 'add',
+      '.submit': 'Register',
+    };
+
+    Status status = await HttpHelper().doMultipartRequest(
+      registerUri,
+      data,
+      queryType: queryType,
+    );
+
+    //Since this is not a official endpoint the response code is always 200
+    //Here we check the response body for certain keyword to find out if the registration was complete
+    if (status.body == null) {
+      return Status(
+        status: 500,
+        error:
+            'No response, open an issue here: https://github.com/openfoodfacts/openfoodfacts-dart/issues/new',
+      );
+    }
+    if (status.body!.contains('loggedin')) {
```

Comment:
Perhaps guard against `status.body` being `null`

Reply:
he's doing so by returning on line 491 I think 👍

Reply:
Exactly, but it is probably more understandable if I connect it to the if else row below it